internal/request/http.go

```diff
@@ -58,7 +77,14 @@ func (h http) bearerToken() string {
 	return strings.ReplaceAll(h.Header.Get("Authorization"), "Bearer ", "")
 }
 
+func (h http) isNonJwtBearerToken() bool {
+	return len(h.bearerToken()) > 0 && !h.isJwtToken()
+}
+
 func (h http) getAuthType() authType {
+	if h.isNonJwtBearerToken() {
```

Comment:
Honestly, since cases are increasing, I would go for something like this:

```go
switch {
case isJwtToken():
  return bearerBased
case len(h.bearerToken()) > 0:
  return bearerBased
case h.TLS != nil:
  return certificateBased
default:
  return anonymousBased
}
```

With this, if it's not a JWT but the bearer token is there, definitely it's a bearer-based authentication so we can avoid putting an additional function (`isNonJwtBearerToken`), WDYT?

Reply:
makes sense, simplified it further, LMK if this looks better.